### PR TITLE
Fixes gh-697 Enhance SqsAutoConfiguration to use an available ObjectM…

### PR DESCRIPTION
**Pull request by mmaeller**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description
See https://github.com/awspring/spring-cloud-aws/discussions/697#discussioncomment-6368312

## :bulb: Motivation and Context
Please see https://github.com/awspring/spring-cloud-aws/discussions/697

## :green_heart: How did you test it?
See extended unit tests.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] I updated reference documentation to reflect the change
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: spring-cloud-aws-autoconfigure/src/test/java/io/awspring/cloud/autoconfigure/sqs/SqsAutoConfigurationTest.java [hunk 1]
```diff
@@ -123,44 +127,99 @@ void customSqsClientConfigurer() {
...
+			var factory = context.getBean(SqsMessageListenerContainerFactory.class);
+			assertThat(factory).hasFieldOrProperty("errorHandler").extracting("asyncMessageInterceptors").asList()
+				.isEmpty();
+			assertThat(factory).extracting("containerOptionsBuilder").asInstanceOf(type(ContainerOptionsBuilder.class))
+				.extracting(ContainerOptionsBuilder::build)
+				.isInstanceOfSatisfying(ContainerOptions.class, options -> {
+					assertThat(options.getMaxConcurrentMessages()).isEqualTo(10);
+					assertThat(options.getMaxMessagesPerPoll()).isEqualTo(10);
+					assertThat(options.getPollTimeout()).isEqualTo(Duration.ofSeconds(10));
+					assertThat(options.getMessageConverter()).isInstanceOfSatisfying(
```
Comment (tomazfernandes) [inline spring-cloud-aws-autoconfigure/src/test/java/io/awspring/cloud/autoconfigure/sqs/SqsAutoConfigurationTest.java:194]: Same here, please let's try to simplify this a bit.

**Review by tomazfernandes — CHANGES_REQUESTED** [review]
Sorry for the delay @mmaeller! Looks good, but there's an issue with the formatting that's creating lots of diffs.

Can you please fix that?

Thanks!

**Comment by postalservice14** [comment]
@tomazfernandes I made the requested changes in a new PR: https://github.com/awspring/spring-cloud-aws/pull/906 - I really would like this released and it seems the original author hasn't had time. :-)

**Comment by mmaeller** [comment]
Hi @postalservice14, Thank you for the adjustments.
And sorry to @tomazfernandes, somehow the mails reached my spam folder :(

**Pull request closed without merging**